docs/impact_health.rst

```diff
@@ -0,0 +1,37 @@
+###########################
+astropy's Impact and Health
+###########################
+
+The figures on this page give a sense of the level of impact astropy has on the astronomy community and research in the field, as well as the amount of development effort contributed over time (the codebase's "health"). A major positive trend is that citations to astropy, and by proxy its usage in research, has been consistently growing since The Astropy Project was created. However, the amount of developer resources has remained small, with a limited number of volunteer maintainers met with an increasing amount of development responsibility.
+
```

Comment:
Since this is effectively a kind of abstract, maybe add a sentence that we'd like help, e.g., "Hence, we welcome any help that you can give!"

Reply:
Good idea! I added the line in [0e164df](https://github.com/astropy/astropy/pull/16822/commits/0e164df73e95118d352f286dd92bb1603a2329d2)